### ISSUE TITLE
Terminating app due to uncaught exception 'NSInvalidArgumentException'

### ISSUE THREAD

**Issue by jimmaye** [launchdarkly/ios-client#82]
Hi

Sometimes when running the client in the iOS simulator (iPhone 6s, iOS10.2), I get this crash - usually during launch of the application (build+run from Xcode):
```
2017-01-24 11:56:34.263 app-name[30637:2191253] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Invalid type in JSON write (UIViewAnimationBlockDelegate)'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001126f9d4b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x000000010cca421e objc_exception_throw + 48
	2   CoreFoundation                      0x00000001127632b5 +[NSException raise:format:] + 197
	3   Foundation                          0x000000010e2fc964 _writeJSONValue + 668
	4   Foundation                          0x000000010e352232 ___writeJSONObject_block_invoke + 226
	5   CoreFoundation                      0x0000000112680dc6 __65-[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:]_block_invoke + 102
	6   CoreFoundation                      0x0000000112680cca -[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:] + 202
	7   Foundation                          0x000000010e352026 _writeJSONObject + 430
	8   Foundation                          0x000000010e2fc84d _writeJSONValue + 389
	9   Foundation                          0x000000010e2fcdb9 ___writeJSONArray_block_invoke + 132
	10  CoreFoundation                      0x00000001126890b2 __53-[__NSArrayM enumerateObjectsWithOptions:usingBlock:]_block_invoke + 114
	11  CoreFoundation                      0x0000000112688f6f -[__NSArrayM enumerateObjectsWithOptions:usingBlock:] + 239
	12  Foundation                          0x000000010e2fcabf _writeJSONArray + 330
	13  Foundation                          0x000000010e2fc8e4 _writeJSONValue + 540
	14  Foundation                          0x000000010e2fc674 -[_NSJSONWriter dataWithRootObject:options:error:] + 124
	15  Foundation                          0x000000010e2fc554 +[NSJSONSerialization dataWithJSONObject:options:error:] + 333
	16  LaunchDarkly                        0x000000010e82b341 -[LDRequestManager performEventRequest:] + 405
	17  LaunchDarkly                        0x000000010e82696f -[LDClientManager syncWithServerForEvents] + 216
	18  LaunchDarkly                        0x000000010e82a5b6 -[LDPollingManager eventPoll] + 101
	19  libdispatch.dylib                   0x0000000113d190cd _dispatch_client_callout + 8
	20  libdispatch.dylib                   0x0000000113cf516c _dispatch_continuation_pop + 1091
	21  libdispatch.dylib                   0x0000000113d0a1de _dispatch_source_latch_and_call + 195
	22  libdispatch.dylib                   0x0000000113d02d79 _dispatch_source_invoke + 1098
	23  libdispatch.dylib                   0x0000000113cfa385 _dispatch_root_queue_drain + 720
	24  libdispatch.dylib                   0x0000000113cfa059 _dispatch_worker_thread3 + 123
	25  libsystem_pthread.dylib             0x00000001140c8712 _pthread_wqthread + 1299
	26  libsystem_pthread.dylib             0x00000001140c81ed start_wqthread + 13
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

Besides that, everything works great. I'm at the newest pod version (2.1.1).

Edit: Screenshot inserted 
![screen shot 2017-01-24 at 12 04 54](https://cloud.githubusercontent.com/assets/6573169/22244871/892f3e02-e22d-11e6-875d-5fda1d2ee954.png)

**Comment by jimmaye** [comment]
Update:
Also happens on devices, but only for debug builds it seems.

I dug through the codebase, and it seems like this commit could be the culprit: [https://github.com/launchdarkly/ios-client/commit/ed6cee2db1ffa9a7979efe5f3ae8645aa3da7a24](url).
Here BOOLs where changed to NSObject, but the property is still assigned, not retained, which Is causing the crash.
Change the following:
```
@property (nonatomic, assign) NSObject * __nonnull value;
@property (nonatomic, assign) NSObject * __nonnull isDefault;

```

to
```
@property (nonatomic, strong) NSObject * __nonnull value;
@property (nonatomic, strong) NSObject * __nonnull isDefault;
```

..and we're good to go it seems. 

Please advice if/when this is a worthy fix - you need to have multivariation feature toggles to provoke the error it seams (where the default value can't be parsed to a BOOL). I'll see if I can submit this as a PR.

**Comment by justinucd** [comment]
Hi - thank you for the detailed issue report!  We're looking into this now.

**Comment by jimmaye** [comment]
@justinucd I submitted a PR for you if needed: [https://github.com/launchdarkly/ios-client/pull/83](url) :-)

**Comment by arun251** [comment]
Fixed in version 2.1.3. Thanks again for the PR.
